tests/unit/test__helpers.py

```diff
@@ -1036,3 +1036,665 @@ def test_default_isolation_and_merge_options_isolation_unspecified(self):
         )
         result = self._callFUT(default, merge)
         self.assertEqual(result, expected)
+
+
+class Test_interval(unittest.TestCase):
+    def _callFUT(self, *args, **kw):
+        from google.cloud.spanner_v1._helpers import _make_value_pb
+
+        return _make_value_pb(*args, **kw)
+
+    def test_basic_interval(self):
+        from google.protobuf.struct_pb2 import Value
+        from google.cloud.spanner_v1 import Type
+        from google.cloud.spanner_v1 import TypeCode
+        from google.cloud.spanner_v1._helpers import Interval
+
+        interval = Interval(months=14, days=3, nanos=43926789000123)
+        field_type = Type(code=TypeCode.INTERVAL)
+        value_pb = self._callFUT(interval)
+        self.assertIsInstance(value_pb, Value)
+        self.assertEqual(value_pb.string_value, "P1Y2M3DT12H12M6.789000123S")
+
+    def test_months_only(self):
+        from google.protobuf.struct_pb2 import Value
+        from google.cloud.spanner_v1 import Type
+        from google.cloud.spanner_v1 import TypeCode
+        from google.cloud.spanner_v1._helpers import Interval
+
+        interval = Interval(months=10, days=0, nanos=0)
+        field_type = Type(code=TypeCode.INTERVAL)
+        value_pb = self._callFUT(interval)
+        self.assertIsInstance(value_pb, Value)
+        self.assertEqual(value_pb.string_value, "P10M")
```

Comment:
nit: Could we have a single test instead where we maintain the actual and expected values array and loop over it.

Reply:
done